### PR TITLE
Samueleceroni/warehouse/stock implementation

### LINKED ISSUE #30: Default quantity unit of measure in Article
Since we already decided to associate only one unit of measure to each article I would like to make this association strong and explicit by putting a field in Article where we decide which unit of measure is used by it.
Otherwise I was thinking about associate each article to a type of unit of measure (eg Volume, weight, units etc.) and to implement a function in Unit of measure to check if the uom is of the correct category (eg if kilogram is weight, Liter is volum etc.). I think this second solution lets us more free to add new Unit of measure (or subunit of measure like grams or decagram) in the future.
Please let me know what you think about this, I wouldn't like to "solve this by myself".
Thanks :kiss:

### LINKED ISSUE #42: Quantity is broken.
When adding or removing something it cast the result to BigDecimal. After this, when i call equals on two quantities where the first has an Integer representation of 1 and the second has a BigDecimal representation of 1, they won't ever be equal one another.

### PR DESCRIPTION
**Pull request by samueleceroni**
implementation of stock impl and tests
fixed #42 
fixed #30

### CODE REVIEW

**Comment by GiackAloZ** [comment]
Can you add the `fixed #42` into pull request description?

File: src/test/java/nwoolcan/model/brewery/warehouse/StockImplTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+     * Initialize structures.
+     */
+    @Before
+    public void init() {
+        expDate = new Date();
+        stock = new StockImpl(ID, ARTICLE, expDate);
+        stock1 = new StockImpl(ID, ARTICLE);
+        try {
+            Thread.sleep(TEN);
+        } catch (Exception ex) {
```
Comment (Zeegomo) [inline src/test/java/nwoolcan/model/brewery/warehouse/StockImplTest.java]: uops, there's a try-catch here...

**Comment by samueleceroni** [comment]
Suggest how to do it without try catch

File: src/test/java/nwoolcan/model/brewery/warehouse/StockImplTest.java (hunk 1 above)
Comment (Zeegomo) [reply src/test/java/nwoolcan/model/brewery/warehouse/StockImplTest.java]:
```suggestion
        Result.ofChecked(() -> Thread.sleep(TEN))
```

Comment (samueleceroni) [reply src/test/java/nwoolcan/model/brewery/warehouse/StockImplTest.java]: to fix this, i need you to Add the version of Results.ofChecked() which accept void as return type of the lambda

Comment (samueleceroni) [reply src/test/java/nwoolcan/model/brewery/warehouse/StockImplTest.java]: Done

**Pull request merged**